frontend/src/Components/Modals/SaveStateModal.tsx

```diff
@@ -20,33 +20,33 @@ const SaveStateModal: FC = () => {
     const saveState = () => {
         const csrftoken = simulateAPIClick()
         if (store.configStore.checkIfInSavedState(stateName)) {
-            setErrorMessage("State name duplicate. Please rename.")
-            setOpenError(true);
+            // setErrorMessage("State name duplicate. Please rename.")
+            // setOpenError(true);
 
             // Keep these for now. For the next PR i will make "udpate state" into a new button, so not allowing duplicate state name in "save state" dialog.
 
-            // fetch(`${process.env.REACT_APP_QUERY_URL}state`, {
-            //     method: `PUT`,
-            //     credentials: "include",
-            //     headers: {
-            //         'Accept': 'application/x-www-form-urlencoded',
-            //         'Content-Type': 'application/x-www-form-urlencoded',
-            //         'X-CSRFToken': csrftoken || '',
-            //         "Access-Control-Allow-Origin": 'https://bloodvis.chpc.utah.edu',
-            //         "Access-Control-Allow-Credentials": "true",
-            //     },
-            //     body: JSON.stringify({ old_name: stateName, new_name: stateName, new_definition: store.provenance.exportState(false) })
-            // }).then(response => {
-            //     if (response.status === 200) {
-            //         onSuccess(false);
-            //     } else {
-            //         response.text().then(error => {
-            //             onFail(response.statusText);
-            //         })
-            //     }
-            // }).catch(error => {
-            //     onFail(error.toString())
-            // })
+            fetch(`${process.env.REACT_APP_QUERY_URL}state`, {
+                method: `PUT`,
+                credentials: "include",
+                headers: {
+                    'Accept': 'application/x-www-form-urlencoded',
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                    'X-CSRFToken': csrftoken || '',
+                    "Access-Control-Allow-Origin": 'https://bloodvis.chpc.utah.edu',
+                    "Access-Control-Allow-Credentials": "true",
+                },
+                body: JSON.stringify({ old_name: stateName, new_name: stateName, new_definition: store.provenance.exportState(false) })
+            }).then(response => {
+                if (response.status === 200) {
+                    onSuccess(false);
+                } else {
+                    response.text().then(error => {
+                        onFail(response.statusText);
+                    })
+                }
+            }).catch(error => {
+                onFail(error.toString())
+            })
```

Comment:
This looks good, but what if we want to open a state, modify it, and then save it as a new one?

Reply:
I am addressing everything with states in the new PR (haven't started by on my todo list this week.)

Reply:
#186